pykeepass/kdbx_parsing/common.py

```diff
@@ -116,43 +135,33 @@ def compute_key_composite(password=None, keyfile=None):
         password_composite = b''
     # hash the keyfile
     if keyfile:
-        # try to read XML keyfile
-        try:
-            with open(keyfile, 'r') as f:
-                tree = etree.parse(f).getroot()
-                version = tree.find('Meta/Version').text
-                data_element = tree.find('Key/Data')
-                if version.startswith('1.0'):
-                    keyfile_composite = base64.b64decode(data_element.text)
-                elif version.startswith('2.0'):
-                    # read keyfile data and convert to bytes
-                    keyfile_composite = bytes.fromhex(data_element.text.strip())
-                    # validate bytes against hash
-                    hash = bytes.fromhex(data_element.attrib['Hash'])
-                    hash_computed = hashlib.sha256(keyfile_composite).digest()[:4]
-                    assert hash == hash_computed, "Keyfile has invalid hash"
-        # otherwise, try to read plain keyfile
-        except (etree.XMLSyntaxError, UnicodeDecodeError):
+        if isinstance(keyfile, bytes):
```

Comment:
See the save method, where it checks whether the file has `hasattr(filename, "write")` maybe the same could be done with the read attribute. BytesIO has this btw.

Reply:
In this case i'd need to change it from bytes to BytesIO. Not sure we want to enforce this in apps using pykeepass? NO strong opinion here, up to the reviewers i guess.